### DISCUSSION TITLE
Filtering props: client/server mismatch

### DISCUSSION THREAD

**Discussion by mirandalambert** [vercel/next.js#18560] [Help] [answered]
I'm attempting to do some filtering of `props` and conditional rendering on the client side only, but I constantly get rendering errors and warnings about client and server being out of sync after hydration. I'm very green when it comes to React. How should I go about manipulating `props` on the client side only? I've created tiny [demo repo](https://github.com/LukeLambert/nextjs-client-side-test) to illustrate the issue. Any help is greatly apprecaited!

**Comment by jamesmosier** [comment, marked as answer]
You could remove the `if (typeof window !== "undefined") {` [from here](https://github.com/LukeLambert/nextjs-client-side-test/blob/e47e875919aefb071abe09637975e0ad03bd9975/pages/index.js#L6) and also have the filtering happen on the server also! You are getting the mismatch because of that conditional. Also be sure to not reassign a value to `movies` as you are, best practice would be to do something like:

```js
export default function Home({ movies: initialMovies }) {
  const movies = initialMovies.map(/* your filtering logic */)
```

---

Otherwise if you really only wanted to filter on the client you could do:

```js
const [movies, setMovies] = useState(props.movies)
useEffect(() => {
  setMovies(/* your sorted/filtered list */)
}, [props.movies])
```

**Reply by mirandalambert** [reply]
`useEffect` works perfectly. The real app (nothing like the demo) will be deployed to static hosting, and I do need some functionality exclusively on the client. Thank you for your help!
